### DISCUSSION TITLE
ZBM can’t import pool

### DISCUSSION THREAD

**Discussion by mitosu-me** [zbm-dev/zfsbootmenu#319] [Questions]
I tried install ZBM with gentoo but after installation ZBM cannot find import pool.

I fallow the guide on gentoo wiki but I found out that in ZBM wiki Void Linux installation guide there are some step isn't in the gentoo guide `zpool set cachefile=/etc/zfs/zpool.cache zroot`, `zpool set bootfs=zroot/ROOT/void zroot` and `zfs set org.zfsbootmenu:commandline="ro quiet" zroot/ROOT`, is this three step mandatory to make ZBM work?

**Comment by zdykstra** [comment]
* `zpool set cachefile=/etc/zfs/zpool.cache zroot` helps speed up the import in your booted OS by caching some data for ZFS
* `zpool set bootfs=zroot/ROOT/void zroot` just tells ZFSBootMenu what OS you'd like to boot by default. If that's not set, you'll just get a menu to pick the environment.
* `org.zfsbootmenu:commandline="ro quiet" zroot/ROOT` is a basic default kernel commandline for your OS

With out knowing how ZFSBootMenu was installed or what specific errors you received when trying to boot, I can't really say what might/might not have happened with your Gentoo install.

**Comment by mitosu-me** [comment]
Just failed again, but I found the problem, since I combined several guide together I missed a session in void install guide
```
zpool export zroot
zpool import -N -R /mnt zroot
zfs mount zroot/ROOT/void
zfs mount zroot/home
```

I mainly follow this guide https://wiki.gentoo.org/wiki/User:Fearedbliss/Installing_Gentoo_Linux_On_ZFS, and he use
```
zpool create -f -o ashift=12 -o cachefile= -O compression=lz4 -O atime=off -m none -R /mnt/gentoo tank /dev/nvme0n1p4
zfs create -o mountpoint=/ tank/os
zfs create -o mountpoint=/home tank/home
```
Since he didn't use `canmount=noauto` and I messed a session, I installed entire system on `/mnt/gentoo`.

I feel I'm extremely dumb.

All good now, I will try again after days.
